src/react-serializer.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+
+'use strict';
+
+const prettyFormat = require('pretty-format');
+
+const { ReactElement } = prettyFormat.plugins;
+const reactElement = Symbol.for('react.element');
+
+function getReactComponentSerializer() {
+  let renderer;
+  try {
+    renderer = require('react-test-renderer'); // eslint-disable-line import/no-extraneous-dependencies
+  } catch (error) {
+    if (error.code === 'MODULE_NOT_FOUND') {
+      throw new Error(
+        `Failed to load optional module "react-test-renderer". ` +
+          `If you need to compare React elements, please add "react-test-renderer" to your ` +
+          `project's dependencies.\n` +
+          `${error.message}`
+      );
+    }
+    throw error;
+  }
+  return value => renderer.create(value).toJSON();
+}
+
+const reactSerializer = {
+  test: (value: any) => value && value.$$typeof === reactElement,
+  print: (value: any) => {
+    const reactComponentSerializer = getReactComponentSerializer();
+    return reactComponentSerializer(value);
+  },
+  diffOptions: (valueA: any, valueB: any) => {
+    const prettyFormatOptions = { plugins: [ReactElement], min: true };
+    return {
+      aAnnotation: prettyFormat(valueA, prettyFormatOptions),
+      bAnnotation: prettyFormat(valueB, prettyFormatOptions),
+    };
+  },
+};
```

Comment:
Code from `src/index.js` moved out to create a react component serializer object - this is currently the only default serializer